File: backend/src/main/java/com/back/back9/domain/exchange/entity/Exchange.java
package com.back.back9.domain.exchange.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "exchange")
public class Exchange {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String symbol;

    private LocalDateTime candleTime;

    private String open;
    private String high;
    private String low;
    private String close;
    private String volume;
    private String timestamp;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
    }

    public LocalDateTime getCandleTime() {
        return candleTime;
    }

    public void setCandleTime(LocalDateTime candleTime) {
        this.candleTime = candleTime;
    }

    public String getOpen() {
        return open;
    }

    public void setOpen(String open) {
        this.open = open;
    }

    public String getHigh() {
        return high;
    }

    public void setHigh(String high) {
        this.high = high;
    }

    public String getLow() {
        return low;
    }

    public void setLow(String low) {
        this.low = low;
    }

    public String getClose() {
        return close;
    }

    public void setClose(String close) {
        this.close = close;
    }

    public String getVolume() {
        return volume;
    }

    public void setVolume(String volume) {
        this.volume = volume;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/exchange/entity/Exchange.kt
package com.back.back9.domain.exchange.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Index
import jakarta.persistence.Table
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity
@Table(
    name = "exchange",
    indexes = [Index(name = "idx_symbol_candletime", columnList = "symbol, candleTime")]
)
data class Exchange(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(nullable = false)
    var symbol: String,

    @Column(nullable = false)
    var candleTime: LocalDateTime,

    @Column(precision = 18, scale = 8, nullable = false)
    var open: BigDecimal,

    @Column(precision = 18, scale = 8, nullable = false)
    var high: BigDecimal,

    @Column(precision = 18, scale = 8, nullable = false)
    var low: BigDecimal,

    @Column(precision = 18, scale = 8, nullable = false)
    var close: BigDecimal,

    @Column(precision = 18, scale = 8, nullable = false)
    var volume: BigDecimal,

    @Column(nullable = false)
    var timestamp: Long
)